### PR TITLE
Extend sign API to support hex string

### PR DESCRIPTION
**Pull request by ytx1991**
### Purpose:
To simplify the Chialisp code using sign API, we need to support signing hex message.

### Current Behavior:
The message will be regarded as a normal string.

### New Behavior:
If the is_hex = true then the message will be convert to bytes

### Testing Notes:
unit test

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1313,8 +1313,11 @@ async def sign_message_by_address(self, request) -> EndpointResult:
         :return:
         """
         puzzle_hash: bytes32 = decode_puzzle_hash(request["address"])
+        is_hex = request.get("is_hex", False)
+        if isinstance(is_hex, str):
+            is_hex = bool(is_hex)
         pubkey, signature = await self.service.wallet_state_manager.main_wallet.sign_message(
-            request["message"], puzzle_hash
+            request["message"], puzzle_hash, request.get("is_hex", is_hex)
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:1320]: shouldn't `request.get("is_hex", is_hex)` be just `is_hex` ?

**Review by trepca — CHANGES_REQUESTED** [review]

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -1362,7 +1368,9 @@ async def sign_message_by_id(self, request) -> EndpointResult:
                 return {"success": False, "error": f"NFT for {entity_id.hex()} doesn't exist."}
 
             assert isinstance(selected_wallet, NFTWallet)
-            pubkey, signature = await selected_wallet.sign_message(request["message"], target_nft)
+            pubkey, signature = await selected_wallet.sign_message(
+                request["message"], target_nft, request.get("is_hex", False)
```
Comment (trepca) [inline chia/rpc/wallet_rpc_api.py:1372]: this needs to be sanitized too

File: tests/wallet/test_wallet.py [hunk 3]
```diff
@@ -847,6 +848,18 @@ async def test_sign_message(
             puzzle.get_tree_hash(),
             G2Element.from_bytes(bytes.fromhex(response["signature"])),
         )
+        # Test hex string
+        message = "0123456789ABCDEF"
+        response = await api_0.sign_message_by_address(
+            {"address": encode_puzzle_hash(ph, "xch"), "message": message, "is_hex": True}
```
Comment (trepca) [inline tests/wallet/test_wallet.py:854]: please add tests with badly formed `is_hex` parameter too

**Review by trepca — APPROVED** [review]

**Comment by altendky** [comment]
Note that discussions need to be resolved.

**Pull request merged**